File: john-cena.m
#import <Foundation/Foundation.h>
///////////////////////////////////
// How to build and execute
//
// Build:
// clang -framework Foundation -o john-cena john-cena.m
//
// ./john-cena
//

int main() {
	NSArray* args = [[NSProcessInfo processInfo] arguments];
	NSFileHandle* stdout = [NSFileHandle fileHandleWithStandardOutput];
	NSData* data = nil;
	if ([args count] == 1) {
		data = [@"\n" dataUsingEncoding: NSUTF8StringEncoding];
	} 

	while (true) {
		[stdout writeData:data];
	}
}

File: john-cena.ml
(* 
 	How to build and execute 
 	
 	Build:
	ocamlopt -o john-cena john-cena.ml
	

	./john-cena

*)

let flood str = 
        while true do
        ignore (Printf.printf "%s\n" str);
        done;;

let () = 
        match Sys.argv with
        | _ -> flood ""
